packages/compiler/test/e2e/cli/cli.e2e.ts

```diff
@@ -0,0 +1,185 @@
+  const result = await execCli(args, { cwd });
+  if (result.exitCode === 0) {
+    throw new Error(`Cli succeeded but expected failure: ${result.stdio}`);
+  }
+  return result;
+}
+
+export interface ExecResult {
+  exitCode: number;
+  stdout: string;
+  stderr: string;
+  stdio: string;
+  proc: ChildProcess;
+}
+export async function execAsync(
+  command: string,
+  args: string[],
+  options: SpawnOptions
+): Promise<ExecResult> {
+  const child = spawn(command, args, options);
+
+  return new Promise((resolve, reject) => {
+    child.on("error", (error) => {
+      reject(error);
+    });
+    const stdio: Buffer[] = [];
+    const stdout: Buffer[] = [];
+    const stderr: Buffer[] = [];
+    child.stdout?.on("data", (data) => {
+      stdout.push(data);
+      stdio.push(data);
+    });
+    child.stderr?.on("data", (data) => {
+      stderr.push(data);
+      stdio.push(data);
+    });
+
+    child.on("exit", (exitCode) => {
+      resolve({
+        exitCode: exitCode ?? -1,
+        stdout: Buffer.concat(stdout).toString(),
+        stderr: Buffer.concat(stderr).toString(),
+        stdio: Buffer.concat(stdio).toString(),
+        proc: child,
+      });
+    });
+  });
+}
+
+async function cleanOutputDir(scenarioName: string) {
+  const dir = resolvePath(getScenarioDir(scenarioName), "tsp-output");
+  await rm(dir, { recursive: true, force: true });
+}
+describe("cli", () => {
+  it("shows help", async () => {
+    const { stdout } = await execCliSuccess(["--help"], {
+      cwd: getScenarioDir("simple"),
+    });
+    expect(stdout).toContain("tsp <command>");
+    expect(stdout).toContain("tsp compile <path>       Compile TypeSpec source.");
+    expect(stdout).toContain("tsp format <include...>  Format given list of TypeSpec files.");
+  });
+
+  describe("compiling spec with warning", () => {
+    it("logs warning and succeed", async () => {
+      const { stdout } = await execCliSuccess(["compile", "."], {
+        cwd: getScenarioDir("warn"),
+      });
+      expect(stdout).toContain("main.tsp:5:8 - warning deprecated: Deprecated: Deprecated");
+      expect(stdout).toContain("Found 1 warning.");
+    });
+
+    it("logs warning as error(and fail) when using --warn-as-error", async () => {
+      const { stdout } = await execCliFail(["compile", ".", "--warn-as-error"], {
+        cwd: getScenarioDir("warn"),
+      });
+      expect(stdout).toContain("main.tsp:5:8 - error deprecated: Deprecated: Deprecated");
+      expect(stdout).toContain("Found 1 error.");
+    });
+  });
+
+  describe("compiling with an emitter", () => {
+    beforeEach(async () => {
+      await cleanOutputDir("with-emitter");
+    });
+
+    it("emits output", async () => {
+      const { stdout } = await execCliSuccess(["compile", ".", "--emit", "./emitter.js"], {
+        cwd: getScenarioDir("with-emitter"),
+      });
+      expect(stdout).toContain("Compilation completed successfully.");
```

Comment:
Is now a good time to point out how I don't like that this logs anything 😂

Reply:
you think it should log nothing at all?

Reply:
Yeah, I feel like it's fine to complete normally without any output when everything went fine. No output means no problems. Anyway, not related to this PR, just thought of it when I read this test.